app/src/main/java/com/android/sample/model/recipe/RecipesViewModel.kt

```diff
@@ -80,6 +80,8 @@ class RecipesViewModel(
   override val categories: StateFlow<List<String>>
     get() = _categories
 
```

Comment:
**General Cleanliness:**  
   Overall, the code changes are clean and align well with the new filtering logic. It’s great to see how seamlessly the new functionality integrates into the existing ViewModel. LGTM!

1. **`isFilterUsed` StateFlow Addition:**  
   Adding `isFilterUsed` is a great touch! It’s a clean and efficient way to track whether filters are applied. Nicely implemented with `checkIfFilterUsed()` updates.

2. **Removing Price Range References:**  
   Smart move cleaning up `priceRange` references to align with the updated `Filter` class. It keeps the logic focused and consistent.

3. **Filter Updates (`updateCategory`, `updateTimeRange`, etc.):**  
   Good call ensuring `isFilterUsed` is updated after each filter change. This makes the code responsive to user actions and ensures the state reflects the current filter configuration.

4. **`fetchRandomRecipes` Logic:**  
   Replacing `searchByCategory` with `filterSearch` is a solid update. The `isFilterUsed` check is a thoughtful addition to ensure the backend query is only made when needed.

5. **Resetting Filters (`resetFilters`):**  
   Nice touch resetting `isFilterUsed` along with the filters. It ensures no residual state issues when users clear filters.

Reply:
Thanks m8! 👍